### ISSUE TITLE
Document (link) nested template approach in amp-list and amp-form docs

### ISSUE THREAD

**Issue by cy137** [ampproject/amphtml#17552]
## What's the issue?

The correct way to handle a form within a list in unintuitive and the docs/examples are out of date. It appears that:

- the runtime accepts nested templates ([example](https://github.com/ampproject/amphtml/blob/master/examples/amp-list-with-form.amp.html#L38))
- the validator does not [issue 16056](https://github.com/ampproject/amphtml/issues/16056)
- the [current docs](https://www.ampproject.org/docs/troubleshooting/validation_errors#disallowed-tag-ancestor) agree with the validator

## How do we reproduce the issue?
1. create an amp-list that contains a form with 
```
<div submit-success>
    <template type="amp-mustache">
    </template>
  </div>
  <div submit
```
2. Fail validation

The current workaround is to name the template and reference it that way.
`<div submit-success template="template-name"></div>`
but this isn't well documented.

## What browsers are affected?

All

## Which AMP version is affected?

1534366245210

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @choumx Do you have any updates?

**Comment by choumx** [comment]
This caveat is now documented in amp-mustache: https://github.com/ampproject/amphtml/blob/master/extensions/amp-mustache/amp-mustache.md#nested-templates

Perhaps we should have pointers in amp-list's and amp-form's documentation?
